src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -899,10 +936,12 @@ ExecuteTaskListExtended(RowModifyLevel modLevel, List *taskList,
 	ParamListInfo paramListInfo = NULL;
 
 	/*
-	 * The code-paths that rely on this function do not know how to execute
-	 * commands locally.
+	 * If current transaction accessed local placements and task list includes
+	 * tasks that should be executed locally (accessing any of the local placements),
+	 * then we should error out as it would cause inconsistencies across the
+	 * remote connection and local execution
```

Comment:
> Is the following equivalent to what is on the patch?

seems equivalent, I will apply this change

Reply:
however, if we detect that we cannot use local execution, then we assign all the tasks to `remoteTaskList` basically. Hence, there `execution->localTaskList` would always be empty as they are executed via a different code path

Reply:
Can we somehow avoid going over each task with `ErrorIfRemoteTaskExecutionOnLocallyAccessedNode `?

Reply:
We should simply do following to prevent unnecessary loop :
```C

	if (TransactionAccessedLocalPlacement)
	{
		ErrorIfRemoteTaskExecutionOnLocallyAccessedNode(taskList);
	}
```